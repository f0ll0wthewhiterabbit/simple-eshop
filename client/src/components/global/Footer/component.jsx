import React from 'react'
import { Grid } from '@material-ui/core'
import PropTypes from 'prop-types'

import { MAIN_PAGE_PATH } from '../../../constants'
import {
  Wrapper,
  Info,
  SiteLink,
  SocialLinksWrapper,
  SocialLinksContainer,
  InfoWrapper,
  InfoHeading,
  InfoList,
  InfoItem,
  SocialLinksList,
  SocialLinksItem,
  SocialLink,
  SocialIconInstagram,
  SocialIconFacebook,
  SocialIconTwitter,
  SocialIconYoutube,
  FooterLink,
  Copyright,
} from './styles'

const Footer = ({ isAdmin, isAuthenticated }) => {
  return (
    <Wrapper>
      {!isAdmin && isAuthenticated && (
        <InfoWrapper maxWidth="lg">
          <Grid container spacing={3}>
            <Grid item xs={12} sm={6} md={3}>
              <InfoHeading variant="h4">About Us</InfoHeading>
              <InfoList>
                <InfoItem>About Us</InfoItem>
                <InfoItem>Community</InfoItem>
                <InfoItem>Jobs</InfoItem>
                <InfoItem>Shipping</InfoItem>
                <InfoItem>Contact Us</InfoItem>
              </InfoList>
            </Grid>
            <Grid item xs={12} sm={6} md={3}>
              <InfoHeading variant="h4">Customer Care</InfoHeading>
              <InfoList>
                <InfoItem>Search</InfoItem>
                <InfoItem>Privacy Policy</InfoItem>
                <InfoItem>2019 Lookbook</InfoItem>
                <InfoItem>Shipping &amp; Delivery</InfoItem>
                <InfoItem>Gallery</InfoItem>
              </InfoList>
            </Grid>
            <Grid item xs={12} sm={6} md={3}>
              <InfoHeading variant="h4">Our Services</InfoHeading>
              <InfoList>
                <InfoItem>Free Shipping</InfoItem>
                <InfoItem>Free Returnes</InfoItem>
                <InfoItem>Our Franchising</InfoItem>
                <InfoItem>Terms and conditions</InfoItem>
                <InfoItem>Privacy Policy</InfoItem>
              </InfoList>
            </Grid>
            <Grid item xs={12} sm={6} md={3}>
              <InfoHeading variant="h4">Information</InfoHeading>
              <InfoList>
                <InfoItem>Payment methods</InfoItem>
                <InfoItem>Times and shipping costs</InfoItem>
                <InfoItem>Product Returns</InfoItem>
                <InfoItem>Shipping methods</InfoItem>
                <InfoItem>Conformity of the products</InfoItem>
              </InfoList>
            </Grid>
          </Grid>
        </InfoWrapper>
      )}

      <SocialLinksWrapper>
        <SocialLinksContainer maxWidth="lg">
          <SocialLinksList>
            <SocialLinksItem>
              <SocialLink href="https://www.instagram.com/itechart_official">
                <SocialIconInstagram className="instagram" />
                <span className="social-links__title">Instagram</span>
              </SocialLink>
            </SocialLinksItem>
            <SocialLinksItem>
              <SocialLink href="https://twitter.com/itechart">
                <SocialIconTwitter className="twitter" />
                <span className="social-links__title">Twitter</span>
              </SocialLink>
            </SocialLinksItem>
            <SocialLinksItem>
              <SocialLink href="https://www.youtube.com/user/iTechArt">
                <SocialIconYoutube className="youtube" />
                <span className="social-links__title">Youtube</span>
              </SocialLink>
            </SocialLinksItem>
            <SocialLinksItem>
              <SocialLink href="https://www.facebook.com/iTechArt.Group">
                <SocialIconFacebook className="facebook" />
                <span className="social-links__title">Facebook</span>
              </SocialLink>
            </SocialLinksItem>
          </SocialLinksList>

          <Info variant="subtitle1" color="textSecondary" align="center" component="p">
            <FooterLink color="inherit" href="https://www.itechart.by/">
              iTechArt
            </FooterLink>
            &nbsp;Students Lab Vitebsk
          </Info>

          <Copyright>
            {'Â© '}
            <SiteLink to={MAIN_PAGE_PATH}>simple eShop</SiteLink> {new Date().getFullYear()}
          </Copyright>
        </SocialLinksContainer>
      </SocialLinksWrapper>
    </Wrapper>
  )
}

Footer.propTypes = {
  isAdmin: PropTypes.bool.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
}

export default Footer
