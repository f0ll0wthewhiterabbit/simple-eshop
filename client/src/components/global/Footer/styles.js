import styled from 'styled-components'
import { Typography, Container } from '@material-ui/core'
import InstagramIcon from '@material-ui/icons/Instagram'
import FacebookIcon from '@material-ui/icons/Facebook'
import TwitterIcon from '@material-ui/icons/Twitter'
import YouTubeIcon from '@material-ui/icons/YouTube'
import { Link } from 'react-router-dom'

export const Wrapper = styled.footer`
  margin-top: auto;
  position: relative;
  z-index: 2;
  background-color: ${props => props.theme.colors.background.footer};
`

export const Info = styled(Typography)`
  font-family: ${props => props.theme.font.family};
  font-size: 15px;
  color: #7b7b7b;
  font-weight: 400;
  line-height: 1.6;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: ${props => props.theme.spacing(1)}px;
`
export const Copyright = styled(Typography)`
  font-family: ${props => props.theme.font.family};
  font-size: 15px;
  color: #7b7b7b;
  font-weight: 400;
  line-height: 1.6;
  text-align: center;
`

export const SiteLink = styled(Link)`
  color: #007bff;
  text-decoration: none;
  transition: all 0.3s;

  &:hover {
    color: #fff;
  }
`

export const SocialLinksWrapper = styled.section`
  background: ${props => props.theme.colors.background.footer};
`

export const SocialLinksContainer = styled(Container)`
  padding-top: ${props => props.theme.spacing(6)}px;
  padding-bottom: ${props => props.theme.spacing(6)}px;
`

export const SocialLinksList = styled.ul`
  margin: ${props => props.theme.spacing(0, 0, 5)};
  padding: 0;
  list-style: none;
  display: flex;
  justify-content: center;
`

export const SocialLinksItem = styled.li`
  margin-right: 20px;

  &:last-child {
    margin-right: 0;
  }

  ${props => props.theme.breakpoints.up('md')} {
    margin-right: 88px;
  }
`

export const SocialLink = styled.a`
  text-decoration: none;
  display: flex;
  align-items: center;

  .social-links__title {
    font-family: ${props => props.theme.font.family};
    font-size: 12px;
    font-weight: 400;
    text-transform: uppercase;
    color: #9f9fa0;
    transition: all 0.3s;
    display: none;

    ${props => props.theme.breakpoints.up('md')} {
      display: inline;
    }
  }

  &:hover {
    .social-links__title {
      color: #fff;
    }

    .instagram {
      color: #2f5d84;
    }

    .twitter {
      color: #5abed6;
    }

    .youtube {
      color: #d12227;
    }

    .facebook {
      color: #39599f;
    }
  }
`

export const SocialIconInstagram = styled(InstagramIcon)`
  font-size: 30px;
  color: #d7d7d7;
  margin-right: 19px;
`

export const SocialIconTwitter = styled(TwitterIcon)`
  font-size: 30px;
  color: #d7d7d7;
  margin-right: 19px;
`

export const SocialIconYoutube = styled(YouTubeIcon)`
  font-size: 30px;
  color: #d7d7d7;
  margin-right: 19px;
`

export const SocialIconFacebook = styled(FacebookIcon)`
  font-size: 30px;
  color: #d7d7d7;
  margin-right: 0;

  && {
    ${props => props.theme.breakpoints.up('md')} {
      margin-right: 19px;
    }
  }
`

export const FooterLink = styled.a`
  color: #007bff;
  text-decoration: none;
  transition: all 0.3s;

  &:hover {
    color: #fff;
  }
`
