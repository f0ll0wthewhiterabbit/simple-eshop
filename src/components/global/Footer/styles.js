import styled from 'styled-components'
import { Typography } from '@material-ui/core'

const Wrapper = styled.footer`
  margin-top: auto;
  position: relative;
  z-index: 2;
  background-color: ${props => props.theme.palette.grey[200]};
  padding: ${props => props.theme.spacing(2, 0, 4)};
`

const Info = styled(Typography)`
  font-size: 0.9rem;
  line-height: 27px;
  display: flex;
  justify-content: center;
  align-items: center;

  ${props => props.theme.breakpoints.up('sm')} {
    line-height: 21px;
    font-size: 1rem;
  }
`

const Logo = styled.img`
  width: 100px;
  height: auto;
  margin-left: 5px;
  margin-right: 5px;
`

export { Wrapper, Info, Logo }
