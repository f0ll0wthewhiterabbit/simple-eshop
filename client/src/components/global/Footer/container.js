import { connect } from 'react-redux'

import Footer from './component'
import { ROLE_ADMIN } from '../../../constants'

const mapStateToProps = state => ({
  isAdmin: state.getIn(['auth', 'user', 'role']) === ROLE_ADMIN,
  isAuthenticated: state.getIn(['auth', 'isAuthenticated']),
})

export default connect(mapStateToProps)(Footer)
