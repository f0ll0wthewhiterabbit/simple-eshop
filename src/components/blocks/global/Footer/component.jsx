import React from 'react'
import { Container, Typography, Link } from '@material-ui/core'

import { Wrapper, Info, Logo } from './styles'

const Footer = () => {
  const copyright = (
    <Typography variant="body2" color="textSecondary" align="center">
      {'Â© '}
      <Link color="inherit" href="/">
        simple eShop
      </Link>{' '}
      {new Date().getFullYear()}
    </Typography>
  )

  return (
    <Wrapper>
      <Container maxWidth="lg">
        <Info variant="subtitle1" color="textSecondary" align="center" component="p">
          <Link color="inherit" href="https://www.itechart.by/">
            <Logo src="/itechart-logo.png" alt="iTechArt" />
          </Link>
          Students Lab Vitebsk
        </Info>
        {copyright}
      </Container>
    </Wrapper>
  )
}

export default Footer
