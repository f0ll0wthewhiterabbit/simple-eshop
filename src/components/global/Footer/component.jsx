import React from 'react'
import { Container, Typography, Link } from '@material-ui/core'

import { MAIN_PAGE_PATH } from '../../../constants'
import { Wrapper, Info, Logo, SiteLink } from './styles'

const Footer = () => {
  const copyright = (
    <Typography variant="body2" color="textSecondary" align="center">
      {'Â© '}
      <SiteLink to={MAIN_PAGE_PATH}>simple eShop</SiteLink> {new Date().getFullYear()}
    </Typography>
  )

  return (
    <Wrapper>
      <Container maxWidth="lg">
        <Info variant="subtitle1" color="textSecondary" align="center" component="p">
          <Link color="inherit" href="https://www.itechart.by/">
            <Logo src="/itechart-logo.png" alt="iTechArt" />
          </Link>
          Students Lab Vitebsk
        </Info>
        {copyright}
      </Container>
    </Wrapper>
  )
}

export default Footer
