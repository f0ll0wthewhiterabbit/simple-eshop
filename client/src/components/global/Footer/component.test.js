import React from 'react'
import { shallow } from 'enzyme'

import Footer from './component'

describe('Footer component', () => {
  it('should render correctly', () => {
    const currentDate = new Date()
    const RealDate = Date
    global.Date = jest.fn(() => new RealDate(currentDate.toISOString()))
    Object.assign(Date, RealDate)

    const wrapper = shallow(<Footer />)

    expect(wrapper).toMatchSnapshot()
  })
})
