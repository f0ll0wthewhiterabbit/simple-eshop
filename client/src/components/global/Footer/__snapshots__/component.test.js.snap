// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render correctly 1`] = `
<styled.footer>
  <WithStyles(ForwardRef(Container))
    maxWidth="lg"
  >
    <Styled(WithStyles(ForwardRef(Typography)))
      align="center"
      color="textSecondary"
      component="p"
      variant="subtitle1"
    >
      <WithStyles(ForwardRef(Link))
        color="inherit"
        href="https://www.itechart.by/"
      >
        <styled.img
          alt="iTechArt"
          src="/itechart-logo.png"
        />
      </WithStyles(ForwardRef(Link))>
      Students Lab Vitebsk
    </Styled(WithStyles(ForwardRef(Typography)))>
    <WithStyles(ForwardRef(Typography))
      align="center"
      color="textSecondary"
      variant="body2"
    >
      Â© 
      <Styled(Link)
        to="/"
      >
        simple eShop
      </Styled(Link)>
       
      2020
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Container))>
</styled.footer>
`;
